### PR TITLE
Port the extended_attributes table to Linux, add support for Linux capabilities

### PR DESCRIPTION
**Pull request by alessandrogario**
This is a port of the following PR from 3.x to 4.x: https://github.com/osquery/osquery/pull/4505

```
osquery> SELECT * FROM extended_attributes WHERE path = "/home/alessandro/test_binary";
+------------------------------+------------------+-------------------------+--------------------------------------------------------------+--------+
| path                         | directory        | key                     | value                                                        | base64 |
+------------------------------+------------------+-------------------------+--------------------------------------------------------------+--------+
| /home/alessandro/test_binary | /home/alessandro | security.capability     | = cap_dac_override,cap_net_admin,cap_net_raw+eip             | 0      |
| /home/alessandro/test_binary | /home/alessandro | user.test_attribute2    | Hello from Trail of Bits!                                    | 0      |
| /home/alessandro/test_binary | /home/alessandro | user.test_attribute     | aGVsbG8sIG9zcXVlcnkh                                         | 1      |
+------------------------------+------------------+-------------------------+--------------------------------------------------------------+--------+
```

### CODE REVIEW

**Comment by theopolis** [comment]
I have a suggestion for an alternate approach to managing source-based libraries in Buck. What do you think of changing the documentation such that Buck open source users first run CMake to setup the source checkouts and patching, then we update the expected paths in Buck to look for a `./build` folder or `./libraries/cmake/source`. This avoids duplicating generated code and avoids duplicating submodule configurations.

**Comment by mike-myers-tob** [comment]
> I have a suggestion for an alternate approach to managing source-based libraries in Buck. What do you think of changing the documentation such that Buck open source users first run CMake to setup the source checkouts and patching, then we update the expected paths in Buck to look for a `./build` folder or `./libraries/cmake/source`. This avoids duplicating generated code and avoids duplicating submodule configurations.

Buck has since been removed from this version of osquery. How can we take a next step on this PR?

**Comment by theopolis** [comment]
The build becomes much easier since we are nearly 100% using the source layer in CMake.

**Comment by theopolis** [comment]
Ping @alessandrogario, are we still interested in merging these features? Is there a strong desire for this data?

**Comment by mike-myers-tob** [comment]
> Ping @alessandrogario, are we still interested in merging these features? Is there a strong desire for this data?

We did write this for one of our clients and would still like to see it merged. I was asking earlier about the next step. It looks like it hasn't been reviewed, but there was a Buck-related suggestion that is I assume no longer applicable. If we rebased this PR and removed the Buck files, would it be eligible for a PR review at that point?

**Comment by directionless** [comment]
> If we rebased this PR and removed the Buck files, would it be eligible for a PR review at that point?

I am broadly interested in this work. The PR does need a refresh though. (sorry)

**Comment by mike-myers-tob** [comment]
> > If we rebased this PR and removed the Buck files, would it be eligible for a PR review at that point?
> 
> I am broadly interested in this work. The PR does need a refresh though. (sorry)

I believe the PR is refreshed and it was just not communicated well. If someone can review/approve, I think this PR is ready.

**Comment by gbedoya** [comment]
+1 on this feature. Of note, file capabilities and process capabilities can be different as a set of capabilities can be preserved across an _execve_. For this reason, I analyze process capabilities so that inheritable capabilities (CapInh) are collected by osquery.
**For example ...** 
```
cat /proc/$pid/status
[...]
CapInh:	0000000000000000
CapPrm:	0000000000000000
CapEff:	0000000000000000
CapBnd:	0000003ff7fdffff
CapAmb:	0000000000000000
NoNewPrivs:	1
Seccomp:	2
```

An approach the parses out /proc table will also allow include **NoNewPrivs** as well as **Seccomp** process attributes. Perhaps, we can extend https://osquery.io/schema/4.5.0/#processes table to include additional attributes? Could someone point me to the https://osquery.io/schema/4.5.0/#processes Linux C/C++ code?

Here's an example of a osquery-go-sdk extension ...
```
osquery> SELECT name, cmdline, seccomp, nonewprivs, capeff, capamb FROM processes_security;
+-----------------+----------------------------------------------+---------+------------+------------------+------------------+
| name            | cmdline                                      | seccomp | nonewprivs | capeff           | capamb           |
+-----------------+----------------------------------------------+---------+------------+------------------+------------------+
| baidu_dns | baidu_dns             | 2       | 1          | 0000000000000000 | 0000000000000000 |
| edgeworker | /usr/local/bin/worker-debug              | 2       | 1          | 0000000000000000 | 0000000000000000 |
```


The PR LGTM aside from renaming extended_attributes to file_capabilties.

**Comment by theopolis** [comment]
I like the idea of another processes_ table for Linux security attributes vs further extending the processes table, but that is not a strong opinion.

One user experience I like is having processes be as cross platform as possible, then each OS’s specific processes-related data shows up in nuanced and specific additional tables. In this scenario the overhead of joining those tables together should be minimal.

**Comment by directionless** [comment]
A downside to using two joined tables, is you can get inconsistencies for rapidly changing data. (This is potentially an issue with processes, we've had customers bump into problems with other vendor solutions joining processes to open sockets)

**Review by theopolis — DISMISSED** [review]

File: osquery/filesystem/tests/posix/xattrs.cpp [hunk 1]
```diff
@@ -0,0 +1,124 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
```
Comment (directionless) [inline osquery/filesystem/tests/posix/xattrs.cpp:2]: old header

File: osquery/tables/system/linux/extended_attributes.cpp [hunk 2]
```diff
@@ -0,0 +1,179 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
```
Comment (directionless) [inline osquery/tables/system/linux/extended_attributes.cpp:2]: beep

File: osquery/tables/system/tests/linux/extended_attributes_tests.cpp [hunk 3]
```diff
@@ -0,0 +1,157 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
```
Comment (directionless) [inline osquery/tables/system/tests/linux/extended_attributes_tests.cpp:2]: boop

**Review by directionless — DISMISSED** [review]

**Review by theopolis — APPROVED** [review]

**Pull request merged**